playwright/_impl/_page.py

```diff
@@ -1274,27 +1261,15 @@ def expect_response(
         urlOrPredicate: URLMatchResponse,
         timeout: float = None,
     ) -> EventContextManagerImpl[Response]:
-        matcher = (
-            None
-            if callable(urlOrPredicate)
-            else URLMatcher(
-                self._browser_context._options.get("baseURL"), urlOrPredicate
-            )
-        )
-        predicate = urlOrPredicate if callable(urlOrPredicate) else None
-
-        def my_predicate(response: Response) -> bool:
-            if matcher:
-                return matcher.matches(response.url)
-            if predicate:
```

Comment:
ditto

Reply:
Addressed in https://github.com/microsoft/playwright-python/pull/2660/commits/0a44a70ae6105dd7a1c24f2174879589e59c126b.